framework/src/bcs/NodalBC.C

```diff
@@ -35,6 +36,18 @@ validParams<NodalBC>()
       "contributions to.  Everything about that variable must match everything "
       "about this variable (the type, what blocks it's on, etc.)");
 
+  // These are the default names for tags, but users will be able to add their own
```

Comment:
yes, I agree these should be provided by a `TagInterface`.

Reply:
I also think for now, we possibly should make these two parameters private and only MOOSE `System` can set them.

Reply:
I think we need some actions to handle these parameters.

Reply:
This is a question we should discuss with other moose members like @permcody